### ISSUE TITLE
If an Rnw file is used as input, a dollar sign in a comment in an R code block shows up in output as a pound-sterling sign

### ISSUE THREAD

**Issue by gsrohde** [yihui/knitr#1952]
---

By filing an issue to this repo, I promise that

- [x] I have fully read the issue guide at https://yihui.org/issue/.
- [x] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('knitr')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('yihui/knitr')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [x] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

---

Example file:
```
%\VignetteEngine{knitr::knitr}
\documentclass{article}

\usepackage[utf8]{inputenc}

%\renewcommand{\hlcom}[1]{\textcolor[rgb]{0.678,0.584,0.686}{\textsl{#1}}}%

\begin{document}

<<list_function>>=
example_initial_values = list(Stem = 3, Leaf = 5)

# You can get a value using the key and the '$' operator:
example_initial_values$Leaf
@

\end{document}
```

If the `\hlcom` command is tweaked by uncommenting the `\renewcommand` line above (which replaces `\textit` by `\textsl`), the problem goes away.

Ultimately, this seems to be a problem with the `alltt` package.  I filed a LaTeX bug report (https://github.com/latex3/latex2e/issues/483) and was advised to add the line
```latex
\usepackage[T1]{fontenc}
```
to the preamble, which also fixes the problem.

This is a satisfactory solution.  Still, I would think it would be a common enough thing for `knitr` users to put $ symbols in R-code comments that it might be worthwhile changing `knitr` in some way so that this problem doesn't arise.  In fact, I'm surprised that no one has pointed out this issue before now.  Perhaps most `knitr` users use `.Rmd` files rather than `.Rnw` files so that this issue doesn't really come up that frequently.

---

`session_info` output:
```
> xfun::session_info('knitr')
R version 4.0.3 (2020-10-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Mojave 10.14.6, RStudio 1.3.1093

Locale: en_US.UTF-8 / en_US.UTF-8 / en_US.UTF-8 / C / en_US.UTF-8 / en_US.UTF-8

Package version:
  evaluate_0.14   glue_1.4.2     
  graphics_4.0.3  grDevices_4.0.3
  highr_0.8       knitr_1.30.9   
  magrittr_2.0.1  markdown_1.1   
  methods_4.0.3   mime_0.9       
  stats_4.0.3     stringi_1.5.3  
  stringr_1.4.0   tools_4.0.3    
  utils_4.0.3     xfun_0.20      
  yaml_2.2.1     
```

**Comment by gsrohde** [comment]
Another solution is to use `xelatex` in place of `pdflatex` to process the `.tex` file produced by `R CMD Sweave`.  (In RStudio, `xelatex` can be selected as the default processor in the build configuration panel.)

**Comment by yihui** [comment]
I vaguely remember in the very early days of **knitr** (circa 2012), I also ran into this problem and tried to let **knitr** add `\usepackage[T1]{fontenc}` to the preamble automatically if it was absent. I don't remember what happened later. Perhaps I gave up this idea and thought it was not a good idea for **knitr** to manipulate user's preamble.

Anyway, it's essentially a LaTeX issue, and I feel it is not trivial to patch it on **knitr**'s side, so I'm going to close it for now (hopefully people could find your solutions in the future via search engines). And many thanks for reporting it to LaTeX developers!
